plugin-scalebar/src/main/java/com/mapbox/pluginscalebar/ScaleBar.java

```diff
@@ -0,0 +1,92 @@
+package com.mapbox.pluginscalebar;
+
+import android.support.annotation.ColorInt;
+import android.support.annotation.NonNull;
+import android.support.annotation.UiThread;
+import android.view.View;
+
+import com.mapbox.android.gestures.StandardScaleGestureDetector;
+import com.mapbox.mapboxsdk.camera.CameraPosition;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.Projection;
+
+/**
+ * Plugin class that show scale bar on mapView and changes scale corresponding to mapView's scale.
+ */
+public class ScaleBar implements MapboxMap.OnScaleListener {
```

Comment:
Implementing `MapboxMap.OnScaleListener` listener on top of this public class also adds `onScaleBegin`, `onScaleEnd` and `onScale` to the public API. It would be nice if we could hide those by using an inner class instead.